imcache.go

```diff
@@ -864,6 +896,12 @@ func (s *Sharded[K, V]) Get(key K) (value V, present bool) {
 	return s.shard(key).Get(key)
 }
 
+// GetAndRemove returns the value for the given key and delete it even if it
+//isn't expired
```

Comment:
```suggestion
// GetAndRemove returns the value for the given key and removes it if it
// exists, otherwise it returns zero value and false.
//
// If it encounters an expired entry, the expired entry is evicted.
```